### PR TITLE
25-validation-error-on-number-fields

### LINKED ISSUE #25: Validation error on number fields
Output json:

```json
{
  "company_name": null,
  "company_address": null,
  "company_website": "lesgranulaines.com",
  "company_phone_number": null,
  "manufacturer_name": null,
  "manufacturer_address": null,
  "manufacturer_website": null,
  "manufacturer_phone_number": null,
  "fertiliser_name": "ALL - PURPOSE NATURAL FERTILISER",
  "registration_number": null,
  "lot_number": "L-1",
  "weight": [],
  "density": null,
  "volume": null,
  "npk": "10-0-4",
  "guaranteed_analysis": [
    { "nutrient": "Total nitrogen (N)", "value": "10", "unit": null },
    { "nutrient": "Soluble potash (K20)", "value": "4", "unit": null },
    { "nutrient": "Organic matter", "value": "61.5", "unit": null }
  ],
  "warranty": null,
  "cautions_en": ["Wash your hands after use.", "Keep away from children."],
  "instructions_en": [
    "1 application per season",
    "Garden: incorporate 150 to 200 grams of Granu-Laines per m2 into first top 10 cm of soil before seeding.",
    "Tomato: incorporate 150 grams underneath each plant.",
    "Cucumber and squash: incorporate 100 grams underneath each plant.",
    "For cultivating in pots or large containers, mix in 50 grams of Granu-Laines per 4 liters of soil.",
    "For any questions, please contact your garden center."
  ],
  "micronutrients_en": [],
  "ingredients_en": [
    { "nutrient": "Ingredients", "value": "100", "unit": null }
  ],
  "specifications_en": [
    { "humidity": "14.8%", "ph": null, "solubility": null }
  ],
  "first_aid_en": [],
  "cautions_fr": [
    "Se laver les mains après chaque utilisation.",
    "Garder hors de la portée des enfants."
  ],
  "instructions_fr": [
    "1 seule application par saison",
    "Pour votre potager, incorporer 150 à 200g de Granu-Laines par m2 de sol.",
    "Tomate: incorporer 150g sous chaque plant.",
    "Concombre et courge: incorporer 100g sous chaque plante.",
    "Pour la culture en pot ou en bac, incorporer 50g par 4 litres de terre.",
    "Et en cas de besoin, renseignez-vous à votre centre jardin."
  ],
  "micronutrients_fr": [],
  "ingredients_fr": [
    { "nutrient": "Ingrédients", "value": "100", "unit": null }
  ],
  "specifications_fr": [
    { "humidity": "14.8%", "ph": null, "solubility": null }
  ],
  "first_aid_fr": []
}

```

Note that in specifications the value of humidity is "14.8%". This causes a parsing error in the datastore.

As discussed with the team, the best approach seems to be enforcing the float or int on number fields and handling the parsing exception in the pipeline.

### PR DESCRIPTION
**Pull request by Endlessflow**
Alright. This PR introduces proper unit tests for the Inspection parsing in `test_inspection.py`.

I also made some fix in `inspection.py` that should make everything more robust and prevent complete failure when a field is invalid. The parsing process now tries to handle the invalid fields and in the event that the field can not be handled we default to null.

@snakedye made a good point that eventually we probably want to find a way to be able to tell the difference between an actually empty value and a parsing error but for now I took the liberty to default to None for both.

@k-allagbe you can let me know if you still get issues with the datastore with this fix

closes #25

### CODE REVIEW

File: pipeline/inspection.py [hunk 1]
```diff
@@ -40,9 +48,13 @@ class Specification(BaseModel):
 
     @field_validator('humidity', 'ph', 'solubility', mode='before', check_fields=False)
     def convert_specification_values(cls, v):
```
Comment (k-allagbe) [inline pipeline/inspection.py:50]: Same here. ph is expected to be a number, not a string.

File: pipeline/inspection.py [hunk 2]
```diff
@@ -19,19 +19,27 @@ class NutrientValue(BaseModel):
 
     @field_validator('value', mode='before', check_fields=False)
     def convert_value(cls, v):
```
[the remark this answers is not in the document]
Comment (Endlessflow) [reply pipeline/inspection.py:21]: I also thought that bool would be implicit. 
But try removing that case and the tests start to fail.
You have to explicitly filter out boolean. 
My best guess is that True is implicitly cast to 1 and False is implicitly cast to 0.

[the remark this answers is not in the document]
Comment (k-allagbe) [reply pipeline/inspection.py:21]: I see. I just learned that bool is a subclass of int. In that case, `type(v) in (int, float)` should do it properlly.
```python
    def convert_value(cls, v):
        if type(v) in (int, float):
            return float(v)
        if type(v) is str:
            return extract_first_number(v)

```

I think type() is more appropriate to check types than isinstance(). What do you think?

**Review by snakedye — APPROVED** [review]
LGTM

**Pull request merged**